app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -60,7 +60,7 @@ def puppet_ca(f, can_override, override)
     select_f f, :puppet_ca_proxy_id, proxies, :id, :name,
              { :include_blank => blank_or_inherit_f(f, :puppet_ca_proxy),
                :disable_button => can_override ? _(INHERIT_TEXT) : nil,
-               :disable_button_enabled => override && !explicit_value?(:puppet_ca_proxy_id),
+               :disable_button_enabled => params["inherit_all"] || (override && !explicit_value?(:puppet_ca_proxy_id)),
```

Comment:
not sure this is the best solution to the issue, but using `@instance_variable` would be a bit better than modifying params.